components/button/button-icon.js

```diff
@@ -110,18 +110,18 @@ class ButtonIcon extends ButtonMixin(VisibleOnAncestorMixin(RtlMixin(LitElement)
 				aria-expanded="${ifDefined(this.ariaExpanded)}"
 				aria-haspopup="${ifDefined(this.ariaHaspopup)}"
 				aria-label="${this.ariaLabel ? this.ariaLabel : ifDefined(this.text)}"
-				?autofocus="${ifDefined(this.autofocus)}"
+				?autofocus="${this.autofocus}"
 				class="d2l-label-text"
 				?disabled="${this.disabled}"
 				form="${ifDefined(this.form)}"
 				formaction="${ifDefined(this.formaction)}"
 				formenctype="${ifDefined(this.formenctype)}"
 				formmethod="${ifDefined(this.formmethod)}"
-				formnovalidate="${ifDefined(this.formnovalidate)}"
+				?formnovalidate="${this.formnovalidate}"
 				formtarget="${ifDefined(this.formtarget)}"
 				name="${ifDefined(this.name)}"
 				title="${ifDefined(this.text)}"
-				type="${this.type}">
+				type="${this._getType()}">
```

Comment:
It was complaining because `type` could have been anything but is only allowed to be one of three things. Interestingly, it's able to infer from the `_getType` method I wrote that it can only be valid... smart!